plugins/inputs/firehose/firehose.go

```diff
@@ -0,0 +1,241 @@
+	tlsConf *tls.Config
+
+	once sync.Once
+	Log  telegraf.Logger
+
+	listener net.Listener
+	server   http.Server
+
+	parser telegraf.Parser
+	acc    telegraf.Accumulator
+}
+
+func (*Firehose) SampleConfig() string {
+	return sampleConfig
+}
+
+func (f *Firehose) Gather(_ telegraf.Accumulator) error {
+	return nil
+}
+
+func (f *Firehose) SetParser(parser telegraf.Parser) {
+	f.parser = parser
+}
+
+func (f *Firehose) Init() error {
+	if f.ServiceAddress == "" {
+		f.ServiceAddress = ":8080"
+	}
+	if len(f.Paths) == 0 {
+		f.Paths = []string{"/telegraf"}
+	}
+	if f.ReadTimeout < config.Duration(time.Second) {
+		f.ReadTimeout = config.Duration(time.Second * 5)
+	}
+	if f.WriteTimeout < config.Duration(time.Second) {
+		f.WriteTimeout = config.Duration(time.Second * 5)
+	}
+
+	var err error
+	f.tlsConf, err = f.ServerConfig.TLSConfig()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Start starts the http listener service.
+func (f *Firehose) Start(acc telegraf.Accumulator) error {
+	f.acc = acc
+
+	var err error
+	if f.tlsConf != nil {
+		f.listener, err = tls.Listen("tcp", f.ServiceAddress, f.tlsConf)
+	} else {
+		f.listener, err = net.Listen("tcp", f.ServiceAddress)
+	}
+	if err != nil {
+		return err
+	}
+
+	f.server = http.Server{
+		Addr:         f.ServiceAddress,
+		Handler:      f,
+		ReadTimeout:  time.Duration(f.ReadTimeout),
+		WriteTimeout: time.Duration(f.WriteTimeout),
+		TLSConfig:    f.tlsConf,
+	}
+
+	go func() {
+		if err := f.server.Serve(f.listener); err != nil {
+			if !errors.Is(err, net.ErrClosed) {
+				f.Log.Errorf("starting server failed: %v", err)
+			}
+		}
+	}()
+
+	f.Log.Infof("listening on %s", f.listener.Addr().String())
+
+	return nil
+}
+
+// Stop cleans up all resources
+func (f *Firehose) Stop() {
+	err := f.server.Shutdown(context.Background())
+	if err != nil {
+		f.Log.Infof("shutting down server failed: %v", err)
+	}
+}
+
+func (f *Firehose) ServeHTTP(res http.ResponseWriter, req *http.Request) {
+	if !slices.Contains(f.Paths, req.URL.Path) {
+		res.WriteHeader(http.StatusNotFound)
+		return
+	}
+
+	requestID := req.Header.Get("x-amz-firehose-request-id")
+	r := &request{req: req}
+
+	if err := r.validate(); err != nil {
+		f.Log.Errorf("validation failed: %v", err)
+		if err = r.sendResponse(res); err != nil {
+			f.Log.Errorf("sending response to request %q failed: %v", requestID, err)
+		}
+		return
+	}
+
+	if err := r.authenticate(f.AccessKey); err != nil {
+		f.Log.Errorf("authentication failed: %v", err)
+		if err = r.sendResponse(res); err != nil {
+			f.Log.Errorf("sending response to request %q failed: %v", requestID, err)
+		}
+		return
+	}
+
+	data, err := r.decodeData()
+	if err != nil {
+		f.Log.Errorf("decode base64 data from request %q failed: %v", requestID, err)
+		if err = r.sendResponse(res); err != nil {
+			f.Log.Errorf("sending response to request %q failed: %v", requestID, err)
+		}
+		return
+	}
+
+	var metrics []telegraf.Metric
+	for _, bytes := range data {
```

Comment:
Please don't use `bytes` as variable name as this is also the name of a built-in Go library. How about `sample` , `msg` or `record`?